File: jmlparser-jml-tools/src/test/java/com/github/jmlparser/jml2java/PatternTest.java
package com.github.jmlparser.jml2java;

import com.github.javaparser.StaticJavaParser;
import com.github.javaparser.ast.expr.NameExpr;
import com.github.javaparser.ast.stmt.ExpressionStmt;
import com.github.javaparser.ast.stmt.IfStmt;
import com.github.jmlparser.utils.Pattern;
import org.junit.jupiter.api.Test;

/**
 * @author Alexander Weigl
 * @version 1 (11.10.22)
 */
class PatternTest {

    @Test
    void matchIf() {
        final var cond = new NameExpr("_");
        final var then = new ExpressionStmt();
        final var other = new ExpressionStmt();
        final var ifPattern = new Pattern<>(new IfStmt(cond, then, other));
        ifPattern.addPlaceholder(cond, "c");
        ifPattern.addPlaceholder(then, "t");
        ifPattern.addPlaceholder(other, "o");

        var candidate = StaticJavaParser.parseStatement("if(a<b+1) print(a); else print(b);");
        var result1 = ifPattern.match(candidate);
        System.out.println(result1);
    }
}

File: tools/jml2java/src/test/kotlin/io/github/jmltoolkit/jml2java/PatternTest.kt
package io.github.jmltoolkit.jml2java

import io.github.jmltoolkit.utils.Pattern
import org.junit.jupiter.api.Test

/**
 * @author Alexander Weigl
 * @version 1 (11.10.22)
 */
internal class PatternTest {
    @Test
    fun matchIf() {
        val cond = com.github.javaparser.ast.expr.NameExpr("_")
        val then = com.github.javaparser.ast.stmt.ExpressionStmt()
        val other = com.github.javaparser.ast.stmt.ExpressionStmt()
        val ifPattern = Pattern(com.github.javaparser.ast.stmt.IfStmt(cond, then, other))
        ifPattern.addPlaceholder(cond, "c")
        ifPattern.addPlaceholder(then, "t")
        ifPattern.addPlaceholder(other, "o")

        val candidate = com.github.javaparser.StaticJavaParser.parseStatement("if(a<b+1) print(a); else print(b);")
        val result1 = ifPattern.match(candidate)
        println(result1)
    }
}